go-controller/pkg/node/gateway_init.go

```diff
@@ -563,3 +563,26 @@ func (nc *DefaultNodeNetworkController) updateGatewayMAC(link netlink.Link) erro
 	return nil
 
 }
+
+func (nc *DefaultNodeNetworkController) getGatewayBridgeName(nodeAddr net.IP) (string, error) {
+	if config.OvnKubeNode.Mode == types.NodeModeDPUHost {
+		gwIntf, err := getInterfaceByIP(nodeAddr)
+		if err != nil {
+			return "", err
+		}
+		config.Gateway.Interface = gwIntf
+	}
+	_, gatewayIntf, err := getGatewayNextHops()
+	if err != nil {
+		return "", err
+	}
+	ifAddrs, err := getNetworkInterfaceIPAddresses(gatewayIntf)
```

Comment:
is this required? I don't think it is getting used as we only return the bridge name

Reply:
yes, you are right. We dont need this it seems. I will test and update the PR accordingly. Thanks for such a nice suggestion :)

Reply:
closing due to change in approach.